cpp/src/arrow/compute/kernels/util_internal.h

```diff
@@ -161,6 +164,21 @@ int64_t CopyNonNullValues(const Datum& datum, T* out) {
   return n;
 }
 
+//--------------------------------------------------------------------------
+// An internal function to create an empty array, chucked array, record batch and table.
+
+Result<std::shared_ptr<Array>> CreateEmptyArray(std::shared_ptr<DataType> type,
+                                                MemoryPool* memory_pool);
```

Comment:
1) Is there a reason these functions are in `compute/kernels/util_internal.h` rather than public APIs? These could go in the respective public headers, such as `array/util.h`, `chunked_array.h`, etc.

2) We generally use "Make" rather than "Create" when naming factory functions (e.g. "MakeArrayOfNull"), can you reuse the same convention here?

Reply:
1. Not really. We can move these functions to their public APIs.
2. Yes, I can rename them. They would be MakeEmptyArray, MakeEmptyChukedArray, MakeEmptyRecordBatch and MakeEmptyTable.